### PR TITLE
99 strict

### LINKED ISSUE #99: Review compiler settings
Are there any additional options that should be activated by default?

### PR DESCRIPTION
**Pull request by ISO50**
close #99

### CODE REVIEW

File: src/types/OAuthConfig.ts [hunk 1]
```diff
@@ -11,4 +11,11 @@ type OAuthConfig = {
   queryParams?: {};
 };
 
-export { OAuthConfig };
+type TokenCacheOAuthConfig = OAuthConfig & {
+  tokenInfoEndpoint: string; // mandatory for TokenCache
```
Comment (bzums) [inline src/types/OAuthConfig.ts]: `tokenInfoEndpoint` can then be removed from `OAuthConfig`, it is not needed there anymore.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# authmosphere
+# authmosphere {🌍}
```
Comment (bzums) [inline README.md]: Niceeee! 🚀 :)

File: src/express-tooling.ts [hunk 3]
```diff
@@ -103,9 +102,17 @@ function handleOAuthRequestMiddleware(options: MiddlewareOptions): RequestHandle
       return next();
     }
 
-    const accessToken = extractAccessToken(getHeaderValue(req, AUTHORIZATION_HEADER_FIELD_NAME));
+    const authHeader = getHeaderValue(req, AUTHORIZATION_HEADER_FIELD_NAME);
+
+    if (!authHeader) {
```
Comment (bzums) [inline src/express-tooling.ts]: Maybe do explicit checking against `undefined` if some strange oauth implementation (or tests) accept `0` as valid value? ^^

File: src/express-tooling.ts [hunk 4]
```diff
@@ -103,9 +102,17 @@ function handleOAuthRequestMiddleware(options: MiddlewareOptions): RequestHandle
...
+      rejectRequest(res);
+      return;
+    }
+
+    const accessToken = extractAccessToken(authHeader);
     if (!accessToken) {
```
Comment (bzums) [inline src/express-tooling.ts]: see above

File: src/utils.ts [hunk 5]
```diff
@@ -50,7 +53,7 @@ export function getFileData(filePath: string, fileName: string): Promise<string>
  * @param field The name of the field to return
  * @returns {string} The value of the header field
  */
-export function getHeaderValue(req: express.Request, fieldName: string): string {
+export function getHeaderValue(req: Request, fieldName: string): string | undefined {
```
Comment (bzums) [inline src/utils.ts]: wrt line 64: should we return there undefined now rather than empty string?

Comment (bzums) [inline src/utils.ts]: (like in `extractAccessToken`)

File: tslint.json [hunk 6]
```diff
@@ -1,5 +1,8 @@
 {
   "rules": {
+    "exclude": [
+      "./**/*.ts"
```
Comment (bzums) [inline tslint.json]: Isnt this excluding everything?

File: src/express-tooling.ts (hunk 3 above)
Comment (bzums) [inline src/express-tooling.ts]: mh ok thinking about it that is maybe not a good idea, we actually want to check against empty strings here also. So maybe better leave it as is.

File: README.md (hunk 2 above)
Comment (ISO50) [inline README.md]: 😀

File: src/types/OAuthConfig.ts (hunk 1 above)
Comment (ISO50) [inline src/types/OAuthConfig.ts]: ✅

File: src/express-tooling.ts (hunk 3 above)
Comment (ISO50) [inline src/express-tooling.ts]: Then `extractAccessToken` will fail?

File: src/utils.ts (hunk 5 above)
Comment (ISO50) [inline src/utils.ts]: What do you think about this solution? I think `hasOwn` was anyway too much, as we can't asume that from `request`.

**Comment by bzums** [comment]
👍

**Comment by ISO50** [comment]
👍

**Comment by ISO50** [comment]
👎

**Comment by Retro64** [comment]
:+1:

**Comment by bzums** [comment]
👍

**Comment by ISO50** [comment]
👍

**Pull request merged**